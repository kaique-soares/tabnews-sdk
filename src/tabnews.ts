import { TabNewsConfig } from './interfaces';

export class TabNews {
  constructor(readonly config?: TabNewsConfig) {
    if (!!config && !config.credentials) {
      throw new Error(
        `Missing Credentials. Pass it to the constructor like the example below:
        \`new TabNews({
            credential: {
              email: 'your@email.com'
              password: 'your_password'
            }
          );\``,
      );
    }

    this.config = config;
  }
}
