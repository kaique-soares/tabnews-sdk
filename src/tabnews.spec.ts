import { expect, describe, it } from 'vitest';
import { TabNews } from './tabnews';

describe('TabNews', () => {
  it('should instantiate without config', () => {
    expect(new TabNews()).toBeDefined();
  });

  it('should instantiate with config', () => {
    expect(
      new TabNews({
        credentials: {
          email: 'test@email.com',
          password: 'dummy_password',
        },
      }),
    ).toBeDefined();
  });

  it('should not instantiate without credentials', () => {
    expect(() => new TabNews({})).toThrowErrorMatchingSnapshot();
  });
});
