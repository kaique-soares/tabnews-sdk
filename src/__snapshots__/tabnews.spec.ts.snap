// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TabNews > should not instantiate without credentials 1`] = `
"Missing Credentials. Pass it to the constructor like the example below:
        \`new TabNews({
            credential: {
              email: 'your@email.com'
              password: 'your_password'
            }
          );\`"
`;
